discovery/gossiper.go

```
@@ -570,8 +570,26 @@ func New(cfg Config, selfKeyDesc *keychain.KeyDescriptor) *AuthenticatedGossiper
 		futureMsgs:        newFutureMsgCache(maxFutureMessages),
 		quit:              make(chan struct{}),
 		chanPolicyUpdates: make(chan *chanPolicyUpdateRequest),
-		prematureChannelUpdates: lru.NewCache[uint64, *cachedNetworkMsg]( //nolint: ll
+		prematureChannelUpdates: lru.NewCache(
 			maxPrematureUpdates,
+			lru.WithDeleteCallback(
+				func(k uint64, cmsg *cachedNetworkMsg) {
+					// for every network message which is
```

Hmm, I see we've changed approaches. Don't we still want to ensure that the [goroutine created](https://github.com/lightningnetwork/lnd/blob/538723e33beaa289596efb82abd8c79d1df08a59/peer/brontide.go#L1998-L2017) to wait the response is cleaned up as soon as possible? 

A premature channel update is an update for a channel we don't know about. It's of common use in the `itests` due to lack of instant block propagation (one node gets the block first, sends the ann before the other has seen the block. It can even be a zombie edge. 

In the common case, we hear of the block then we can return an error and the goroutine exits. However, if it's a zombie, and stays that way for weeks, then these goroutines will still pile up. Or if it's a nearly valid, but fake channel update, we'll never actually process it.

---

Thinking about it a more, this doest at least restrict the amount of these goroutines waiting for a premature update to be processed to `maxPrematureUpdates`, which currently is `100`.

---

exactly I think that's the cleanest solution

---

also added the safeguard to exit the goroutine after a timeout, I think this combination should be a good temporary fix until we have the actor model in place.

---

removed the goroutine